### PR TITLE
Additional entitlement tests

### PR DESCRIPTION
**Pull request by rabbah**
adding CLI entitlement tests, refactoring guest and test subjects to allow for overriding where needed.

### CODE REVIEW

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 1]
```diff
@@ -29,37 +31,77 @@ import common.TestUtils.TIMEOUT
 import common.Wsk
 import common.WskProps
 import common.WskTestHelpers
-import spray.json.DefaultJsonProtocol._
+import spray.json.DefaultJsonProtocol.StringJsonFormat
```
Comment (markusthoemmes) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: any reason not to do:
```
import spray.json._
import spray.json.DefaultJsonProtocol._
```

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 2]
```diff
@@ -29,37 +31,77 @@ import common.TestUtils.TIMEOUT
...
-
-    val wskadmin = new RunWskAdminCmd {}
-    val otherNamespace = Subject().toString
-    val create = wskadmin.cli(Seq("user", "create", otherNamespace))
-    val otherAuthkey = create.stdout.trim
-    val otherWskProps = WskProps(namespace = otherNamespace, authKey = otherAuthkey)
-    val samplePackage = "samplePackage"
-    val sampleAction = s"$samplePackage/sampleAction"
+    lazy val defaultWskProps = WskProps()
+    lazy val guestWskProps = getAdditionalTestSubject()
```
Comment (markusthoemmes) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: Any particular reason to have those as `lazy`?

Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: see line 71 (for overriding classes).

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala (hunk 1 above)
Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: no - eclipse auto import at work.

Comment (markusthoemmes) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: You can tell it to leave `_` imports alone. I find it more convenient to just import everything for spry.json, so everything "just works".

**Comment by rabbah** [comment]
PG2-44 approved.

**Comment by rabbah** [comment]
PG2-48 approved.

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 3]
```diff
@@ -29,37 +31,76 @@ import common.TestUtils.TIMEOUT
...
+        val wskadmin = new RunWskAdminCmd {}
+        val newSubject = Subject().toString
+        WskProps(
+            namespace = newSubject,
+            authKey = wskadmin.cli(Seq("user", "create", newSubject)).stdout.trim)
+    }
+
+    def disposeAdditionalTestSubject(subject: String) = {
+        val wskadmin = new RunWskAdminCmd {}
+        withClue(s"failed to delete temporary namespace $subject") {
```
Comment (markusthoemmes) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: "temporary subject"?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala [hunk 4]
```diff
@@ -173,31 +216,37 @@ class WskCoreBasicTests
...
+            val stdout = wsk.action.get(fullyQualifiedActionName)(defaultWskProps).stdout
             stdout should include("name")
             stdout should include("parameters")
             stdout should include("limits")
             stdout should include regex (""""key": "a"""")
             stdout should include regex (""""value": "A"""")
+
+            val run = wsk.action.invoke(fullyQualifiedActionName)(defaultWskProps)
+
+            withActivation(wsk.activation, run, initialWait = 1.seconds, totalWait = 60.seconds)({
```
Comment (markusthoemmes) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: Any particular reason why you need to specify a longer totalWait time? Notice that the initialWait is consistent with the default.

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala (hunk 3 above)
Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: > "temporary subject"?

Good one. Want to change it at merge time?

File: tests/src/whisk/core/cli/test/WskEntitlementTests.scala (hunk 4 above)
Comment (rabbah) [inline tests/src/whisk/core/cli/test/WskEntitlementTests.scala]: > Any particular reason why you need to specify a longer totalWait time? Notice that the initialWait is consistent with the default.

No - copy/paste overlooked it.

**Comment by markusthoemmes** [comment]
fixed nits, rebased and merged in ff063cb3636ff108b5e24bf4a4514cec079b3c4e

**Pull request closed without merging**